Document controller (#12)
kuzzleio/sdk-dart: doc/2/controllers/document/m-update/index.md

---
code: true
type: page
title: mUpdate
description: Updates multiple documents
---

# mUpdate

Updates multiple documents.

---

## Arguments

```dart
Future<Map<String, dynamic>> mUpdate(
      String index, String collection, 
      List<Map<String, dynamic>> documents,
      {bool waitForRefresh = false, int retryOnConflict})

> ```

Reviewer: Nitpicking
```suggestion
      {
      bool waitForRefresh = false, 
      int retryOnConflict
})
```
```

| `index`            | <pre>String</pre>                                       | Index                             |
| `collection`       | <pre>String</pre>                                       | Collection                        |
| `documents`        | <pre>List<Map<String, dynamic>></pre> | ArrayList containing the documents to update |
| `retryOnConflict`  | <pre>int</pre><br>(`null`)                | The number of times the database layer should retry in case of version conflict |
| `waitForRefresh`   | <pre>bool</pre><br>(`false`)                | If set to `true`, Kuzzle will wait for the persistence layer to finish indexing |

---

### documents

Each document has the following properties:

| Arguments          | Type                                         | Description                       |
| ------------------ | -------------------------------------------- | --------------------------------- |
| `_id`              | <pre>String</pre>                            | Document ID             |
| `body`             | <pre>Map<String, dynamic></pre> | Document body |

## Return

> A `ConcurrentHashMap<String, ArrayList<Object>>` which has a `successes` and `errors`:

Reviewer: ```suggestion
A `Map<String, dynamic>` which has a `successes` and `errors:
```

---
code: true
type: page
title: mUpdate
description: Updates multiple documents
---

# mUpdate

Updates multiple documents.

---

## Arguments

```dart
Future<Map<String, dynamic>> mUpdate(
      String index, String collection, 
      List<Map<String, dynamic>> documents,
      {
      bool waitForRefresh = false, 
      int retryOnConflict
})

| Arguments          | Type                                                    | Description                       |
| ------------------ | ------------------------------------------------------- | --------------------------------- |
| `index`            | <pre>String</pre>                                       | Index                             |
| `collection`       | <pre>String</pre>                                       | Collection                        |

> | `documents`        | <pre>List<Map<String, dynamic>></pre> | ArrayList containing the documents to update |

Reviewer 2: ```suggestion
| `documents`        | <pre>List<Map<String, dynamic>></pre> | List containing the documents to update |
```

| `retryOnConflict`  | <pre>int</pre><br>(`null`)                | The number of times the database layer should retry in case of version conflict |
| `waitForRefresh`   | <pre>bool</pre><br>(`false`)                | If set to `true`, Kuzzle will wait for the persistence layer to finish indexing |

---

### documents

Each document has the following properties:

| Arguments          | Type                                         | Description                       |
| ------------------ | -------------------------------------------- | --------------------------------- |
| `_id`              | <pre>String</pre>                            | Document ID             |
| `body`             | <pre>Map<String, dynamic></pre> | Document body |

## Return

> A `Map<String, dynamic>` which has a `successes` and `errors:

Reviewer 2: ```suggestion
A `Map<String, dynamic>` which has a `successes` and `errors`:
```
